--- xfmedia-0.9.1/xfmedia.desktop.in.orig	2005-09-05 06:56:41.000000000 +0200
+++ xfmedia-0.9.1/xfmedia.desktop.in	2005-11-30 15:52:48.000000000 +0100
@@ -8,6 +8,6 @@
 TryExec=xfmedia
 Icon=xfmedia
 Terminal=false
-Categories=X-XFCE;Application;AudioVideo;AudioPlayer;VideoPlayer;
+Categories=X-XFCE;AudioVideo;Player;
 StartupNotify=true
 MimeType=application/x-annodex;audio/x-ogg;audio/x-speex;application/x-ogg;video/x-ms-asf;video/x-ms-wmv;video/x-ms-wma;application/vnd.ms-asf;application/x-mplayer2;video/x-ms-asf-plugin;video/x-ms-wvx;video/x-ms-wax;audio/x-aiff;audio/aiff;audio/x-pn-aiff;audio/x-realaudio;audio/basic;audio/x-basic;audio/x-pn-au;audio/x-wav;audio/wav;audio/x-pn-wav;audio/x-pn-windows-acm;video/msvideo;video/x-msvideo;video/quicktime;video/x-quicktime;audio/x-m4a;application/x-quicktimeplayer;video/x-flic;audio/x-pn-realaudio;audio/x-pn-realaudio-plugin;audio/x-real-audio;application/vnd.rn-realmedia;video/mkv;audio/x-8svx;audio/8svx;audio/x-16sv;audio/168sv;image/x-ilbm;image/ilbm;video/x-anim;video/anim;video/mpeg;video/x-mpeg;audio/mpeg2;audio/x-mpeg2;audio/mpeg3;audio/x-mpeg3;audio/mpeg;audio/x-mpeg;audio/x-mpegurl;audio/mpegurl;audio/mp3;audio/x-mp3;application/x-flac;
--- xfmedia-0.8.0/xfmedia.desktop.in~	2006-04-18 09:25:20.000000000 +0200
+++ xfmedia-0.8.0/xfmedia.desktop.in	2006-04-18 09:25:20.000000000 +0200
@@ -6,5 +6,5 @@
 Icon=xfmedia
 Terminal=false
 Type=Application
-Categories=X-XFCE;Application;AudioVideo;AudioPlayer;VideoPlayer
+Categories=X-XFCE;AudioVideo;Player;
 StartupNotify=true
